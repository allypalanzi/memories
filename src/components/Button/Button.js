import React from 'react'
import PropTypes from 'prop-types'
import styles from './Button.scss'

const Button = props => {
  const { onClick, text } = props
  return (
    <button className={styles.button} onClick={onClick}>
      {text}
    </button>
  )
}

Button.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string,
}

export default Button
