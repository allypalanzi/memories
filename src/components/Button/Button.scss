.button {
  border: 0;
  padding: 1em;
  background: #cffdcf;
  margin: .5em;
  border: 1px solid darken(#cffdcf, 10%);
  cursor: pointer;
  transition: background 200ms ease-in-out;
  &:hover {
    background: darken(#cffdcf, 20%);
    transition: background 200ms ease-in-out;
  }
}
