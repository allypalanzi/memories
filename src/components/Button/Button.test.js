import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Button from './Button.js';

describe('<Button/ >', () => {
  const props = {
    onClick: jest.fn(),
    text: 'I am a button, click me'
  }

  const renderedComponent = shallow(<Button {...props} />);

  it('renders correctly', () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
