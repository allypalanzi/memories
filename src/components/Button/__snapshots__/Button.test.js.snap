// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/ > renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    onClick={[Function]}
    text="I am a button, click me"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "I am a button, click me",
      "className": "button",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "I am a button, click me",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am a button, click me",
        "className": "button",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "I am a button, click me",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
